segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexAppendProcessor.java

```diff
@@ -0,0 +1,91 @@
+import lombok.extern.slf4j.Slf4j;
+
+import static io.pravega.segmentstore.contracts.Attributes.EVENT_COUNT;
+import static io.pravega.shared.NameUtils.getIndexSegmentName;
+import static io.pravega.shared.NameUtils.isTransactionSegment;
+import static io.pravega.shared.NameUtils.isTransientSegment;
+import static io.pravega.shared.NameUtils.isUserStreamSegment;
+
+/**
+ * Process incoming Append request for index segment.
+ */
+@Slf4j
+public class IndexAppendProcessor {
+    static final Duration TIMEOUT = Duration.ofMinutes(1);
+    private final StreamSegmentStore store;
+    private final MultiKeyLatestItemSequentialProcessor<String, Long> appendProcessor;
+
+    public IndexAppendProcessor(ScheduledExecutorService indexSegmentUpdateExecutor, StreamSegmentStore store) {
+        this.store = store;
+        appendProcessor = new MultiKeyLatestItemSequentialProcessor<>(this::handleIndexAppend, indexSegmentUpdateExecutor);
+    }
+
+    /**
+     * Appends index segment on a separate thread.
+     * @param segmentName  segment name.
+     * @param indexSegmentEventSize  Index segment event size.
+     *                               Value 0 indicates the segment is not a Stream Segment.
+     */
+    protected void processAppend(String segmentName, long indexSegmentEventSize) {
+        // Not updating index segment for transient and transaction type.
+        if (isTransientSegment(segmentName) || isTransactionSegment(segmentName) || !isUserStreamSegment(segmentName)) {
+            return;
+        }
+        if (indexSegmentEventSize != NameUtils.INDEX_APPEND_EVENT_SIZE) {
+            log.debug("The data received for index segment append is not of desired size. Segment: {}, Actual: {}, Desired: {}",
+                    getIndexSegmentName(segmentName), indexSegmentEventSize, NameUtils.INDEX_APPEND_EVENT_SIZE);
+            return;
+        }
+        appendProcessor.updateItem(segmentName, indexSegmentEventSize);
+    }
+
+    private void handleIndexAppend(String segmentName, long indexSegmentEventSize) {
+        store.getStreamSegmentInfo(segmentName, TIMEOUT)
+                .thenCompose(info -> {
+                    long eventCount = info.getAttributes().get(EVENT_COUNT) != null ? info.getAttributes().get(EVENT_COUNT) : 0;
+                    ByteBufWrapper byteBuff = getIndexAppendBuf(info.getLength(), eventCount);
+                    AttributeUpdateCollection attributes = AttributeUpdateCollection.from(
+                            new AttributeUpdate(EVENT_COUNT, AttributeUpdateType.ReplaceIfGreater, eventCount));
+                    return store.append(getIndexSegmentName(segmentName), byteBuff, attributes, TIMEOUT);
+
+                }).thenAccept(v -> log.trace("Index segment append successful for segment {} ", getIndexSegmentName(segmentName)))
+                .exceptionally(ex -> {
+                    log.warn("Index segment append failed for segment {} due to ", getIndexSegmentName(segmentName), ex);
```

Comment:
A {} is missing in the log which prints the exception

Reply:
No need to add {} for exception.